### DISCUSSION TITLE
\htmlId not working

### DISCUSSION THREAD

**Discussion by nschloe** [KaTeX/KaTeX#3084] [Q&A]
From [the docs](https://katex.org/docs/supported.html) I gather that one can set HTML IDs for reference from elsewhere
```
https://katex.org/docs/supported.html
```
However,
```
a = b \htmlId{test}{x}
```
will just print `htmlId` in red

![screenshot](https://user-images.githubusercontent.com/181628/122250953-651c4800-ceca-11eb-9244-6c0378f3336a.png)

My render options are
```js
katex.render(element.textContent, element, {
  displayMode: false,
  throwOnError: false,
  trust: ["\\htmlId"]
});
```
This is with KaTeX 0.13.11.

Any idea what may be wrong here?

(Btw I added KaTeX to [xdoc](https://github.com/nschloe/xdoc), too.)

**Comment by edemaine** [comment]
[The `trust` option](https://katex.org/docs/options.html) takes a function, not an array. Try this:

```js
katex.render(element.textContent, element, {
  displayMode: false,
  throwOnError: false,
  trust: ({command}) => command === "\\htmlId"
});
```

**Reply by edemaine** [reply]
P.S. xdoc looks cool. It might be worth linking to from https://katex.org/docs/libs.html in a new section on Chrome plugins.

**Comment by nschloe** [comment]
I just found a seemingly unrelated error, this also fixes the `trust` above. It _does_ actually work with a list like
```js
trust: ["\\htmlId"]
```

**Reply by edemaine** [reply]
That "works" because `["\\htmlId"]` is true. It is equivalent to `trust: true`, i.e., trust everything.

**Reply by nschloe** [reply]
I see! Thanks for the clarification.
